src/compiler/compiler.rs

```diff
@@ -504,36 +502,25 @@ pub fn detect_compiler_kind<T>(creator: &T, executable: &str, pool: &CpuPool)
     where T: CommandCreatorSync
 {
     trace!("detect_compiler");
-    let write = pool.spawn_fn(move || -> io::Result<_> {
-        let dir = TempDir::new("sccache")?;
-        let src = dir.path().join("testfile.c");
-        write_file(&src, b"#if defined(_MSC_VER)
+    let test = b"#if defined(_MSC_VER)
 msvc
 #elif defined(__clang__)
 clang
 #elif defined(__GNUC__)
 gcc
 #endif
-")?;
-        Ok((dir, src))
-    });
-    let write = write.chain_err(|| "failed to write temporary file");
+".to_vec();
```

Comment:
Ah yeah I assumed this wasn't too bad as well because the compiler info is typically cached anyway (so should hit this rarely)